lib/node_modules/@stdlib/repl/lib/main.js

```diff
@@ -268,6 +271,35 @@ function REPL( options ) {
 		}
 	}
 
+	/**
+	* Callback invoked upon a readline interface "keypress" event.
+	*
+	* @private
+	* @param {string} chunk - character pressed by the user
+	* @param {Object} key - keypress event information
+	*/
+	function onKeypress( chunk, key ) {
+		var brackets = {
+			'{': '}',
+			'[': ']',
+			'(': ')',
+			'\'': '\'',
+			'"': '"',
+			'`': '`'
+		};
+
+		// For ALT+ENTER keypress, add a new line in multi-line mode:
+		if ( key && key.name === 'return' && key.meta ) {
+			self._rli._insertString( '\n' );
```

Comment:
As you allude to in the OP, we should not be using private methods on the readline instance.

Reply:
My guess is that it does not work with `write`, as writing `\n` may trigger the `line` event.

Reply:
For reference as to how we currently handle multi-line input, see https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/repl/lib/process_line.js and https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/repl/lib/display_prompt.js.

To write a newline, you can write to the `ostream`. I imagine the same could be done for bracket matching.